src/dc_simplify.rs

```diff
@@ -39,12 +30,9 @@ pub unsafe fn dc_simplify_simplify(
     /* create a copy of the given buffer */
     let mut out: *mut libc::c_char;
     let mut temp: *mut libc::c_char;
-    if simplify.is_null() || in_unterminated.is_null() || in_bytes <= 0i32 {
-        return dc_strdup(b"\x00" as *const u8 as *const libc::c_char);
-    }
-    (*simplify).is_forwarded = 0i32;
-    (*simplify).is_cut_at_begin = 0i32;
-    (*simplify).is_cut_at_end = 0i32;
+    self.is_forwarded = 0i32;
+    self.is_cut_at_begin = 0i32;
+    self.is_cut_at_end = 0i32;
```

Comment:
if changing such lines anyway i'd also remove the "i32".

Reply:
I'll try to do it next time. In any case, dc_simplify will need a second pass of refactoring to make it use rusty strings.